tests/migration/migration.go

```diff
@@ -3140,6 +3141,196 @@ var _ = SIGMigrationDescribe("VM Live Migration", decorators.RequiresTwoSchedula
 			Eventually(matcher.ThisMigration(migration)).WithPolling(1 * time.Second).WithTimeout(60 * time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceMigrationRejectedByResourceQuota))
 		})
 	})
+	Context("Virtiofs", decorators.VirtioFS, func() {
+		DescribeTable("should migrate with a shared", func(createConfigVolume func() (string, string, string, []libvmi.Option)) {
+
+			configVolumeName, expectedOutput, testCommand, configVolumeOption := createConfigVolume()
+
+			vmi := libvmifact.NewAlpine(
+				libvmi.WithInterface(libvmi.InterfaceDeviceWithMasqueradeBinding()),
+				libvmi.WithNetwork(v1.DefaultPodNetwork()),
+			)
+
+			for _, opt := range configVolumeOption {
+				opt(vmi)
+			}
+
+			By("Running VMI")
+			vmi = libvmops.RunVMIAndExpectLaunch(vmi, 180)
+
+			By("Logging into the VMI")
+			Expect(console.LoginToAlpine(vmi)).To(Succeed())
+
+			By("Checking mounted ConfigVolume")
+			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: fmt.Sprintf("mount -t virtiofs %s /mnt \n", configVolumeName)},
+				&expect.BExp{R: console.PromptExpression},
+				&expect.BSnd{S: "echo $?\n"},
+				&expect.BExp{R: console.RetValue("0")},
+				&expect.BSnd{S: testCommand},
+				&expect.BExp{R: expectedOutput},
+			}, 200)).To(Succeed())
+
+			By("Starting the migration")
+			migration := libmigration.New(vmi.Name, vmi.Namespace)
+			migration = libmigration.RunMigrationAndExpectToCompleteWithDefaultTimeout(virtClient, migration)
+
+			libmigration.ConfirmVMIPostMigration(virtClient, vmi, migration)
+
+			By("Checking the ConfigVolume is still mounted and data integrity")
+			Expect(console.SafeExpectBatch(vmi, []expect.Batcher{
+				&expect.BSnd{S: testCommand},
+				&expect.BExp{R: expectedOutput},
+			}, 200)).To(Succeed())
+		},
+			Entry("ConfigMap", func() (string, string, string, []libvmi.Option) {
```

Comment:
I'm wondering if functional testing with all types is necessary to check different functionalities or some of them could be converted to unit test. The reason behind my comment is that we are trying to reducing the functional tests when possible

Reply:
+1

Reply:
So, leaving as functional test just the ConfigMap and a PVC?, or just the ConfigMap?

Reply:
So, as we agreed, we will keep functional tests just for a configmap and a PVC. Testing the rest as unit test.

Reply:
done